ElvaRivle/Feature/Sidebar

ElvaRivle opened the pull request:
FEATURE/ElvaRivle/Sidebar

--------------------------------------------------------------------
 MAIN
 * added Sidebar component
 * added temporary route only for sidebar
 * added ant-design icons package
 * added all resources neccessary for sidebar 
--------------------------------------------------------------------
 OTHER
 * added text editor config files to .gitignore
--------------------------------------------------------------------
 TODO
 * add styling to component

ElvaRivle committed:
FEATURE/ElvaRivle/Sidebar

--------------------------------------------------------------------
 MAIN
 * added Sidebar component
 * added temporary route only for sidebar
 * added ant-design icons package
 * added all resources neccessary for sidebar
--------------------------------------------------------------------
 OTHER
 * added text editor config files to .gitignore
--------------------------------------------------------------------
 TODO
 * add styling to component

ElvaRivle committed:
ElvaRivle/Feature/Sidebar

--------------------------------------------------------------------
 MAIN
 * added styling to sidebar
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------

becarselman commented on src/components/sidebar/Sidebar.js:
```diff
@@ -0,0 +1,71 @@
+import Logotype from '../../images/sidebar/Logotype.svg';
+import SampleProfilePicture from '../../images/sidebar/Sample-Profile-Image.png'
+import { useEffect, useState } from "react";
+import {
```
Imports should have different spots.
-React imports
-3party imports
-local imports
-images
-css

becarselman commented on src/components/sidebar/Sidebar.js:
```diff
@@ -0,0 +1,71 @@
+
+      <div className="sidebar-options">
+        <div className="sidebar-option">
+          <HomeOutlined />
```
Can we map this from array of objects. so in case we want to add another item just add item in array of objects

becarselman commented on src/components/sidebar/Sidebar.scss:
```diff
@@ -0,0 +1,104 @@
+  left: 0;
+  top: 0;
+
+  background: linear-gradient(179.32deg, #FCFDFC 8.04%, #E6F2ED 99.41%);
```
move this colors in scss colors file

becarselman commented on src/components/sidebar/Sidebar.scss:
```diff
@@ -0,0 +1,104 @@
+  top: 0;
+
+  background: linear-gradient(179.32deg, #FCFDFC 8.04%, #E6F2ED 99.41%);
+  border-right: 1px solid #EBF4F1;
```
same

becarselman commented on src/components/sidebar/Sidebar.scss:
```diff
@@ -0,0 +1,104 @@
+}
+
+.user-role {
+  font-family: 'Inter', sans-serif;
```
you use same style for font few times. Create mixin in mixins file and reuse it on all places

becarselman commented on src/routing/router.js:
```diff
@@ -1,13 +1,16 @@
       <Routes>
         <Route path="/" element={<Navigate to="/login" />} />
         <Route path="/login" element={<LoginForm />} />
+        <Route path="/sidebar" element={<Sidebar />} />
```
```
<Router>
    <div>
        <Sidebar></Sidebar>
           <Switch>
            <Route exact path={"/"} component={Charts} />
            <Route exact path={"/about"} component={About} />
           </Switch>
        <Footer></Footer>
    </div>
</Router>
```

here is solution for your comment. But we don't want sidebar on login so we need to create 2 sets of routes. protected and auth. On protected routes we will have this side bar.

ElvaRivle replied:
Should we worry about protected routes now or when we properly implement login logic?

becarselman reviewed (commented):
Please add screen shot of sidebar

ElvaRivle commented:
![Sidebar image](https://github.com/becarselman/ant-colony-pharaoh-frontend/assets/46258494/8157a15c-fa7a-44f7-bde2-fe6a3c0dc6a7)

ElvaRivle committed:
ElvaRivle/Feature/Sidebar

--------------------------------------------------------------------
 MAIN
 * added font mixin
 * colors moved to _colors.scss
 * created array of sidebar options and properly displayed it
 * import order fixed
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------

ElvaRivle commented:
ElvaRivle/Feature/Sidebar

--------------------------------------------------------------------
 MAIN
 * added font mixin
 * colors moved to _colors.scss
 * created array of sidebar options and properly displayed it 
 * import order fixed 
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------

ElvaRivle committed:
ElvaRivle/Feature/Sidebar

--------------------------------------------------------------------
 MAIN
 * removed leftover mixin
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------

ElvaRivle commented:
ElvaRivle/Feature/Sidebar

--------------------------------------------------------------------
 MAIN
 * removed leftover mixin
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------

becarselman commented on src/components/sidebar/Sidebar.js:
```diff
@@ -0,0 +1,49 @@
+import SampleProfilePicture from '../../images/sidebar/Sample-Profile-Image.png'
+import './Sidebar.scss'
+
+function Sidebar({ actions }) {
```
use arrow function :D

becarselman commented on src/components/sidebar/Sidebar.js:
```diff
@@ -0,0 +1,49 @@
+      </div>
+
+      <div className="sidebar-options">
+        {sidebarOptions.map(option => {
```
You can move map outside of jsx. Doing that you reduce loading time because app don't need to map each time on render.

becarselman commented on src/components/sidebar/Sidebar.scss:
```diff
@@ -0,0 +1,93 @@
+.sidebar-option p {
+  margin-left: 20px;
+
+  @include font("Gilroy", normal, 500, 14px, 16px, $text-color-secondary);
```
I don't mean this. create mixin for similar staff for example font-family, font-weight

@mixin bold-font {
  font-family: 'Goldplay-Bold';
  font-style: normal;
  font-weight: 600;
} 
as you can se here I use just this 3 stuff and font size I add manualy

becarselman reviewed (changes requested):
Make sure your product look 100% same as design

ElvaRivle commented:
> Make sure your product look 100% same as design

I used icons from Ant design. I guess I should export icons from Figma and use exactly those?

ElvaRivle commented:
> Make sure your product look 100% same as design

I used icons from Ant design. I guess I should export icons from Figma and use exactly those?

ElvaRivle committed:
FEATURE/ElvaRivle/Sidebar

--------------------------------------------------------------------
 MAIN
 * fixed font mixins
 * moved `sidebarOptions.map` outside JSX
--------------------------------------------------------------------
 OTHER
 * Google fonts import URL moved to Sidebar.scss because it didn't load in index.css
 * small bottom margin for sidebar options
--------------------------------------------------------------------

ElvaRivle commented:
FEATURE/ElvaRivle/Sidebar

--------------------------------------------------------------------
 MAIN
 * fixed font mixins
 * moved `sidebarOptions.map` outside JSX
--------------------------------------------------------------------
 OTHER
 * Google fonts import URL moved to Sidebar.scss because it didn't load in index.css
 * small bottom margin for sidebar options
--------------------------------------------------------------------

ElvaRivle committed:
FEATURE/ElvaRivle/Sidebar

--------------------------------------------------------------------
 MAIN
 * protected /sidebar route
--------------------------------------------------------------------
 OTHER
 * for the beginning, only basic "is accessToken present" check is implemented
--------------------------------------------------------------------

ElvaRivle commented:
FEATURE/ElvaRivle/Sidebar

--------------------------------------------------------------------
 MAIN
 * protected /sidebar route
--------------------------------------------------------------------
 OTHER
 * for the beginning, only basic "is accessToken present" check is implemented
--------------------------------------------------------------------

becarselman commented on src/routing/router.js:
```diff
@@ -1,17 +1,24 @@
   return (
     <Router>
       <Routes>
+        <Route element={<ProtectedRoutes />}>
```
It will be good to have sidebar on each protected route. I already put example in some comment so you can look at it. First protected route will be /dashboard and it will include sidebar.

becarselman replied:
My suggestion is to create something like this 
```<AuthRoute path="/login" exact component={Login} />```

becarselman replied:
and if you want protected route just type <ProtectedRoute path="/login" exact component={Login} /> 
and if you want something else you can create for example AdminRoute...

ElvaRivle committed:
ElvaRivle/Feature/Sidebar

--------------------------------------------------------------------
 MAIN
 * created wrapper component protectedRoutes that implements core authN/authZ logic
 * every set of protected routes should be child of that protectedRoutes component
 * created single index route for dashboard routes
--------------------------------------------------------------------
 OTHER
 * created empty Dashboard component
 * default route is not /login, but /dashboard, redirect to login will happen if user not authN/authZ
--------------------------------------------------------------------

ElvaRivle commented:
ElvaRivle/Feature/Sidebar

--------------------------------------------------------------------
 MAIN
 * created wrapper component ProtectedRoutes that implements core authN/authZ logic (simply checking is token present)
 * every set of protected routes should be child of that ProtectedRoutes component
 * created single index route for dashboard routes
--------------------------------------------------------------------
 OTHER
 * created empty Dashboard component
 * default route is not /login, but /dashboard, redirect to login will happen if user not authN/authZ
--------------------------------------------------------------------
